File: ios/AllTranslate/RNSpeechLocales.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RNSpeechLocales, NSObject)

RCT_EXTERN_METHOD(getSupportedLocales:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/AllTranslate/RNSpeechLocales.swift
import Foundation
import Speech

@objc(RNSpeechLocales)
class RNSpeechLocales: NSObject {
  @objc(getSupportedLocales:rejecter:)
  func getSupportedLocales(resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    let identifiers = Array(SFSpeechRecognizer.supportedLocales()).map { $0.identifier }
    resolve(identifiers)
  }
}


